pyveda/vedaset/store/vedabase.py

```diff
@@ -214,10 +212,25 @@ def __del__(self):
 
 
 class VedaBase(H5DataBase):
-
     class _MetaSample(tables.IsDescription):
         vid = tables.StringCol(36)
 
+    def __init__(self, *args, dataset_id=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        if dataset_id is not None:
+            self._root._v_attrs["dataset_id"] = dataset_id
+
+    def __iter__(self):
+        img_iter = self._root.images.iterrows()
+        lbl_iter = self._root.labels.iterrows()
+        return(zip(img_iter, lbl_iter))
```

Comment:
Nice, so a couple things here: 1) nice on initializing the underlying iterable arrays, that means that even if somebody was calling `next` on say the image array, we're resetting the iteration pointers so that each image and label will start at same pointer. 2) `zip` is likely loading all of this data into memory. This is a python convention but there are built-in iteration toolsets that are really handy for doing things like this: checkout the `itertools` module, and specifically something like `zip_longest` if you want to go this route. 3) The usual convention in `__iter__` is to return `self`, what that does is understand to check call the function `__next__` defined on the object we're iterating over (our VedaBase object). Then we have the freedom to define `next` using all the object available to on our object, like for instance the metadata table... see the iterator section [here](https://docs.python.org/3/tutorial/classes.html) for reference, and there are other good explanations of how this pattern works in python on stackoverflow for more in depth